ephios/core/services/notifications/backends.py

```diff
@@ -151,7 +151,7 @@ def send(cls, notification):
             "icon": "/static/ephios/img/ephios-symbol-red.svg",
         }
         if actions := notification.get_actions():
-            payload["url"] = actions[0][1]
+            payload["url"] = f"{actions[0][1]}?fromNotification={notification.pk}"
```

Comment:
Instead of appending this to every <a> tag manually, we could just do a pass over the generated html and fix the href tags. That also links inside paragraphs would be tracked.

Reply:
You are also not checking whether the actions point to the correct url/ephios instance.

Reply:
just fixing the HTML wouldn't be enough, as this line that you are quoting here is not part of the HTML. If you want to avoid the magic string I think wrapping the get_actions() to return the fixed links would be the better approach. I already thought about it but discarded it because I thought its fine this way

Reply:
I do mean the complete rendered E-Mail HTML, not only the message content HTML. Of course parsing the URL again would be redundant, but I think it is worth it to have it separated from normal message URL construction.

Reply:
I still don't really get it, lets talk about this at some point
The notification is only marked as read in the middleware if the notification belongs to the user that is currently logged in, so I see no attack vector there